packages/@aws-cdk/aws-appmesh/lib/virtual-gateway.ts

```diff
@@ -0,0 +1,494 @@
+      resourceName: this.physicalName,
+    });
+  }
+}
+
+function renderHealthCheck(hc: HealthCheck | undefined, pm: PortMapping): CfnVirtualGateway.VirtualGatewayHealthCheckPolicyProperty | undefined {
+  if (hc === undefined) { return undefined; }
+
+  if (hc.protocol === Protocol.TCP && hc.path) {
+    throw new Error('The path property cannot be set with Protocol.TCP');
+  }
+
+  if (hc.protocol === Protocol.GRPC && hc.path) {
+    throw new Error('The path property cannot be set with Protocol.GRPC');
+  }
+
+  const protocol = hc.protocol? hc.protocol : pm.protocol;
+
+  const healthCheck: CfnVirtualGateway.VirtualGatewayHealthCheckPolicyProperty = {
+    healthyThreshold: hc.healthyThreshold || 2,
+    intervalMillis: (hc.interval || cdk.Duration.seconds(5)).toMilliseconds(), // min
+    path: hc.path || ((protocol === Protocol.HTTP || protocol === Protocol.HTTP2) ? '/' : undefined),
+    port: hc.port || pm.port,
+    protocol: hc.protocol || pm.protocol,
+    timeoutMillis: (hc.timeout || cdk.Duration.seconds(2)).toMilliseconds(),
+    unhealthyThreshold: hc.unhealthyThreshold || 2,
+  };
+
+  validateHealthChecks(healthCheck);
+
+  return healthCheck;
+}
+
+/**
+ * Unterface with properties necessary to import a reusable VirtualGateway
+ */
+interface VirtualGatewayAttributes {
+  /**
+   * The name of the VirtualGateway
+   */
+  readonly virtualGatewayName?: string;
+
+  /**
+   * The Amazon Resource Name (ARN) belonging to the VirtualGateway
+   */
+  readonly virtualGatewayArn?: string;
+
+  /**
+   * The Mesh that the VirtualGateway belongs to
+   */
+  readonly mesh?: IMesh;
+
+  /**
+   * The name of the mesh that the VirtualGateway belongs to
+   */
+  readonly meshName?: string;
+}
+
+/**
+ * Used to import a VirtualGateway and read its properties
+ */
+class ImportedVirtualGateway extends VirtualGatewayBase {
+  /**
+   * The name of the VirtualGateway
+   */
+  public readonly virtualGatewayName: string;
+
+  /**
+   * The Amazon Resource Name (ARN) belonging to the VirtualGateway
+   */
+  public readonly virtualGatewayArn: string;
+
+  /**
+   * The Mesh that the VirtualGateway belongs to
+   */
+  public readonly mesh: IMesh;
+
+  constructor(scope: Construct, id: string, props: VirtualGatewayAttributes) {
+    super(scope, id);
+
+    if (props.mesh) {
+      this.mesh = props.mesh;
+    } else if (props.meshName) {
+      if (props.mesh) {
+        throw new Error('Supply either \'mesh\' or \'meshName\', but not both');
+      }
+      this.mesh = Mesh.fromMeshName(this, 'Mesh', props.meshName);
+    } else if (props.virtualGatewayArn) {
+      const meshName = cdk.Fn.select(0, cdk.Fn.split('/', cdk.Stack.of(scope).parseArn(props.virtualGatewayArn).resourceName!));
+      this.mesh = Mesh.fromMeshName(this, 'Mesh', meshName);
+    } else {
+      throw new Error('Supply either \'mesh\' or \'meshName\' or \'virtualGatewayArn\'');
+    }
+    if (props.virtualGatewayArn) {
```

Comment:
Same comment here - since we know that we only have a `fromVirtualGatewayArn` method, a lot of this code in the `if`s can be removed.